### DISCUSSION TITLE
How to customize External Links extension in Symfony 4/5?

### DISCUSSION THREAD

**Discussion by StefanGlaesser** [thephpleague/commonmark#784] [Q&A]
Hello,

currently I'm using Symfony 4 and the twig/markdown-extra with the league/commonmark extension.

I would like to customize the external link behavior as it is described here:
https://commonmark.thephpleague.com/2.1/extensions/external-links/

There is also a note on https://twig.symfony.com/doc/3.x/filters/markdown_to_html.html
"If using Symfony (full-stack), twig/extra-bundle with league/commonmark as your Markdown library you can configure CommonMark extensions. Register the desired extension(s) as a service, then tag the service with twig.markdown.league_extension."

So I created an Service

```
  namespace App\Twig;
  
  use League\CommonMark\Environment\Environment;
  use League\CommonMark\Extension\CommonMark\CommonMarkCoreExtension;
  use League\CommonMark\Extension\ExternalLink\ExternalLinkExtension;
  use League\CommonMark\MarkdownConverter;
  
  class MarkdownExternalLinkExtension
  {
      public function __construct(Environment $environment) {
  
          $config = [
              'external_link' => [
                  'internal_hosts' => 'my-domain.com',
                  'open_in_new_window' => true,
                  'html_class' => 'external-link',
                  'nofollow' => 'external',
                  'noopener' => 'external',
                  'noreferrer' => 'external',
              ]
          ];
  
          $environment = new Environment($config);
          $environment->addExtension(new CommonMarkCoreExtension());
  
          // Add this extension
          $environment->addExtension(new ExternalLinkExtension());
  
          $converter = new MarkdownConverter($environment);
      }
  }
```
and  tagged the service as described:

```
    App\Twig\MarkdownExternalLinkExtension:
        tags:
            - { name: 'twig.markdown.league_extension' }
```

But my registered extension doesn't do anything :(
Has someone implemented in Symfony and can give me a hint?

Best regards,
Stefan

**Comment by colinodell** [comment]
Your custom extension probably needs to `implement ExtensionInterface` and look something like this:

```php
final class MarkdownExternalLinkExtension implements ExtensionInterface
{
    public function register(ConfigurableEnvironmentInterface $environment): void
    {
        // TODO: Do whatever you need to do here
    }
}
```

I'm not super familiar with the inner workings of the Twig integration, but hopefully that points you in the right direction :)

**Reply by StefanGlaesser** [reply]
Hi,
thanks for the idea. I gave it a try, but unfortunately without success.
Best regards,
Stefan
